### PR TITLE
DesktopManager

### PR DESCRIPTION
**Pull request by StMarian**

### CODE REVIEW

File: EasyAccess/EADesktopManager/DesktopHelper.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#include "DesktopHelper.h"
+
+const std::string DesktopHelper::desktopName = "EAHiddenDesktop";
+
+DesktopHelper::DesktopHelper(const std::string& appFullPath)
+    : appFullPath(appFullPath)
+    , log("C:\\EasyAccess\\log.txt")
```
Comment (Kp0c) [inline EasyAccess/EADesktopManager/DesktopHelper.cpp]: TODO: suggest some local path

File: EasyAccess/EADesktopManager/DesktopHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        log.Log("Runapp success");
+        //WaitForSingleObject(app, INFINITE);
+    }
+    else
+    {
+        log.Log("Failed to runapp");
+    }
+
+    // Ctrl-Alt-E
+    if (RegisterHotKey(NULL, 1, MOD_CONTROL | MOD_ALT | MOD_NOREPEAT, 0x45))
```
Comment (Kp0c) [inline EasyAccess/EADesktopManager/DesktopHelper.cpp]: TODO: remove in "production"))

File: EasyAccess/EADesktopManager/DesktopHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
+#include "DesktopHelper.h"
+
+const std::string DesktopHelper::desktopName = "EAHiddenDesktop";
```
Comment (Kp0c) [inline EasyAccess/EADesktopManager/DesktopHelper.cpp]: **CRITICAL!**
desktop name should be combination of user + application, so user can run different apps in different dekstops.

Comment (StMarian) [reply EasyAccess/EADesktopManager/DesktopHelper.cpp]: Sure, forgot about that workflow.

File: EasyAccess/EADesktopManager/DesktopHelper.cpp (hunk 2 above)
Comment (StMarian) [reply EasyAccess/EADesktopManager/DesktopHelper.cpp]: will remove after final testing
don't want to reboot due to small mistakes)

File: EasyAccess/CredentialProvider/Credential.cpp [hunk 4]
```diff
@@ -328,6 +337,11 @@ HRESULT Credential::GetSerialization(__out CREDENTIAL_PROVIDER_GET_SERIALIZATION
                 _events->AppendFieldComboBoxItem(this, FI_COMBOBOX, _applications[i]);
             }
 
+            if (IsWin7())
```
Comment (Kp0c) [inline EasyAccess/CredentialProvider/Credential.cpp]: I think we have to add comment to this workaround :)

File: EasyAccess/EADesktopManager/DesktopHelper.cpp [hunk 5]
```diff
@@ -19,9 +19,10 @@ void DesktopHelper::Start()
 
     //Need to switch thread into context of new desktop to register hotkeys
     SetThreadDesktop(hiddenDesc);
-    if (!SwitchDesktop(hiddenDesc))
+    while (!SwitchDesktop(hiddenDesc))
```
Comment (Kp0c) [inline EasyAccess/EADesktopManager/DesktopHelper.cpp]: is ti OK?

File: EasyAccess/CredentialProvider/Credential.cpp (hunk 4 above)
Comment (StMarian) [reply EasyAccess/CredentialProvider/Credential.cpp]: Oh, sure)

**Pull request merged**